pylint/checkers/typecheck.py

```diff
@@ -806,6 +810,10 @@ def _is_invalid_isinstance_type(arg: nodes.NodeNG) -> bool:
         return False
     if isinstance(inferred, astroid.Instance) and inferred.qname() == BUILTIN_TUPLE:
         return False
+    if isinstance(inferred, bases.UnionType):
```

Comment:
Wondering if we should put this line 806 before the Tuple check for performance reason. Wet finger estimation but I suppose union happens more often (?)

Reply:
The union syntax is pretty new, and I wouldn't think adoption is all that widespread. For example, it's not used anywhere in Pylint code because of compatibility with old versions.

Reply:
Make sense !